saleor/warehouse/availability.py

```diff
@@ -190,6 +199,16 @@ def check_stock_quantity_bulk(
         )
 
         if quantity > 0:
+            if quantity_limit is not None and quantity > quantity_limit:
+                raise ValidationError(
+                    {
+                        "quantity": ValidationError(
+                            "Cannot add more than %d times this item: %s."
+                            "" % (quantity_limit, variant),
```

Comment:
Why not f-string?
```suggestion
                            f"Cannot add more than {quantity_limit} times this item: {variant}."
```

Reply:
C - like (let's say) string interpolation was used in this error before small refactor. Anyway, done.